File: src/test/polyglot/fibonacci.py
print('Fibonacci in Python on GraalVM!')

n = 42

def fibonacci(n):
    if n == 1:
        return 1
    elif n == 2:
        return 1
    else:
        return fibonacci(n-1) + fibonacci(n-2)

def fibonacci_iter(n):
    if n == 1:
        return 1
    elif n == 2:
        return 1
    else:
        a = 1
        b = 1
        for i in range(2, n):
            c = a + b
            a = b
            b = c
        return b

print("{0}'s fibonacci value is {1}".format(n, fibonacci(n)))

File: src/test/polyglot/fibonacci.js
function fibonacci(n) {
    if (n === 1 || n === 2) {
        return 1
    } else {
        return fibonacci(n-1) + fibonacci(n-2)
    }
}

function fibonacci_iter(n) {
    if (n === 1 || n === 2) {
        return 1
    } else {
        let a = 1
        let b = 1
        let c = 0
        for (let i = 2; i < n; i++) {
            c = a + b
            a = b
            b = c
        }
        return c
    }
}

console.log(fibonacci(42))
